crates/uv/src/commands/tool/run.rs

```diff
@@ -695,36 +694,41 @@ async fn get_or_create_environment(
 
     let reporter = PythonDownloadReporter::single(printer);
 
-    // Check if the target is `python`
-    let python_request = if request.is_python() {
-        let target_request = match &request.target {
-            Target::Unspecified(_) => None,
-            Target::Version(_, _, _, version) => Some(PythonRequest::Version(
-                VersionRequest::from_str(&version.to_string()).map_err(anyhow::Error::from)?,
-            )),
-            // TODO(zanieb): Add `PythonRequest::Latest`
-            Target::Latest(_, _, _) => {
-                return Err(anyhow::anyhow!(
-                    "Requesting the 'latest' Python version is not yet supported"
-                )
-                .into());
-            }
-        };
-
-        if let Some(target_request) = &target_request {
-            if let Some(python) = python {
-                return Err(anyhow::anyhow!(
-                    "Received multiple Python version requests: `{}` and `{}`",
-                    python.to_string().cyan(),
-                    target_request.to_canonical_string().cyan(),
-                )
-                .into());
+    // Figure out what Python we're targeting, either explicitly like `uvx python@3`, or via the
+    // -p/--python flag.
+    let python_request = match request {
+        ToolRequest::Python(tool_py_req) => {
```

Comment:
Nit: Generally we don't shorten variable names, I'd just use `tool_request` or `tool_python_request`.